app/src/main/java/com/ayush/weatherapp/main/MainPresenter.java

```diff
@@ -0,0 +1,67 @@
+package com.ayush.weatherapp.main;
+
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.support.annotation.NonNull;
+import com.ayush.weatherapp.main.pojo.Currently;
+import com.ayush.weatherapp.main.pojo.Forecast;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class MainPresenter implements MainContract.Presenter {
+
+  private static ProgressDialog progressDialog;
```

Comment:
Presenter should not contain any view(android.View) objects. This should be in the activity. Context cannot be helped.